docs/sources/k6/next/testing-guides/performance-testing-grpc-services.md

```diff
@@ -0,0 +1,206 @@
+
+#### Unary
+
+Unary calls work the same way as a regular function call: a single request is sent to the server which in turn replies with a single response.
+
+![Unary call](/media/docs/k6-oss/diagram-grpc-unary.png)
+
+#### Server streaming
+
+In server streaming mode, the client sends a single request to the server, which in turn replies with multiple responses.
+
+![Server streaming](/media/docs/k6-oss/diagram-grpc-server.png)
+
+#### Client streaming
+
+The client streaming mode is the opposite of the server streaming mode. The client sends multiple requests to the server, which in turn replies with a single response.
+
+![Client streaming](/media/docs/k6-oss/diagram-grpc-client.png)
+
+#### Bi-directional streaming
+
+In bi-directional streaming mode, both the client and the server may send multiple messages.
+
+![Bi-directional streaming](/media/docs/k6-oss/diagram-grpc-bidirectional.png)
+
+### The `.proto` definition
+
+The messages and services used for gRPC are described in .proto files, containing [Protocol buffers](https://en.wikipedia.org/wiki/Protocol_Buffers), or protobuf, definitions.
+
+The definition file is then used to generate code which can be used by both senders and receivers as a contract for communicating through these messages and services. As the binary format used by gRPC lacks any self-describing properties, this is the only way for senders and receivers to know how to interpret the messages.
+
+For this guide, you can use the `quickpizza.proto` definition available for download on the [QuickPizza demo application GitHub repository](https://raw.githubusercontent.com/grafana/quickpizza/refs/heads/main/proto/quickpizza.proto). For details on how to build your own gRPC proto definition, refer to [the official gRPC docs](https://grpc.io/docs/what-is-grpc/core-concepts/).
+
+```protobuf
+syntax = "proto3";
+option go_package = "pkg/grpc/quickpizza";
+package quickpizza;
+
+service GRPC {
+    rpc Status(StatusRequest) returns (StatusResponse) {}
+    rpc RatePizza(PizzaRatingRequest) returns (PizzaRatingResponse) {}
+}
+
+message StatusRequest {
+}
+
+message StatusResponse {
+    bool ready = 1;
+}
+
+message PizzaRatingRequest {
+    repeated string ingredients = 1;
+    string dough = 2;
+}
+
+message PizzaRatingResponse {
+    int32 stars_rating = 1;
+}
+```
+
+## Write a gRPC performance test with k6
+
+Starting with k6 v0.29.0, you can use a built-in module for gRPC communication. You can find more details for all the methods available on the [k6/net/grpc module documentation](https://grafana.com/docs/k6/<K6_VERSION>/javascript-api/k6-net-grpc/).
+
+### Create the test
+
+The gRPC module is a separate module, available from your test script as `k6/net/grpc`. Before you can use it, you first have to create an instance of the client. Instantiating the client, as well as the `.load` operation, is only available during test initialization, that is, directly in the global scope.
```

Comment:
Here, we say _"`.load` operation"_, but below, we say _"`.load()` function"_. Should we keep them consistent, like `.load()` function?

I also wonder if using a dot prefix is a common practice in our docs. I'd omit the dot for simplicity (i.e., `load()` function). Or, it can be `Client.load()` if we want to keep it verbose and explicit.

Reply:
That's a good point, I think we can just refer to it as `load()` function.